backend/v3/storage/database/repository/settings.go

```diff
@@ -0,0 +1,1106 @@
+}
+
+const eventCreateSettingStmt = `INSERT INTO zitadel.settings` +
+	` (instance_id, organization_id, type, owner_type, label_state, settings, created_at, updated_at)` +
+	` VALUES ($1, $2, $3, $4, $5, $6, $7, $8)` +
+	` RETURNING id, created_at, updated_at`
+
+func (s *settings) Create(ctx context.Context, client database.QueryExecutor, setting *domain.Setting) error {
+	builder := database.NewStatementBuilder(
+		eventCreateSettingStmt,
+		setting.InstanceID,
+		setting.OrganizationID,
+		setting.Type,
+		setting.OwnerType,
+		setting.LabelState,
+		string(setting.Settings),
+		setting.CreatedAt,
+		setting.UpdatedAt)
+
+	return client.QueryRow(ctx, builder.String(), builder.Args()...).Scan(&setting.ID, &setting.CreatedAt, &setting.UpdatedAt)
+}
+
+const createSettingStmt = `INSERT INTO zitadel.settings` +
+	` (instance_id, organization_id, type, owner_type, label_state, settings)` +
+	` VALUES ($1, $2, $3, $4, $5, $6)` +
+	` ON CONFLICT (instance_id, organization_id, type, owner_type) WHERE type != 'label' DO UPDATE SET` +
+	` settings =  EXCLUDED.settings::JSONB `
+
+func createSetting(ctx context.Context, client database.QueryExecutor, setting *domain.Setting, settings any, changes ...database.Change) error {
+	if setting.InstanceID == "" {
+		return ErrMissingInstanceID
+	}
+
+	if setting.OrganizationID == nil {
+		return ErrMissingOrgID
+	}
+
+	// type is always set from the calling function to that functions respective type
+	// the check is left as a fail-safe
+	if setting.Type == domain.SettingTypeUnspecified {
+		return ErrMissingType
+	}
+
+	if setting.OwnerType == domain.OwnerTypeUnspecified {
+		return ErrMissingOwnerType
+	}
+
+	settingJSON, err := json.Marshal(settings)
+	if err != nil {
+		return err
+	}
+
+	builder := database.NewStatementBuilder(
+		createSettingStmt,
+		setting.InstanceID,
+		setting.OrganizationID,
+		setting.Type,
+		setting.OwnerType,
+		// TODO
+		setting.LabelState,
+		string(settingJSON))
+
+	err = database.Changes(changes).Write(builder)
+	if err != nil {
+		return err
+	}
+
+	builder.WriteString(` RETURNING id, created_at, updated_at`)
+
+	return client.QueryRow(ctx, builder.String(), builder.Args()...).Scan(&setting.ID, &setting.CreatedAt, &setting.UpdatedAt)
+}
+
+func (s *settings) Delete(ctx context.Context, client database.QueryExecutor, condition database.Condition) (int64, error) {
+	if err := s.CheckMandatoryCondtions(condition); err != nil {
+		return 0, err
+	}
+
+	var builder database.StatementBuilder
+	builder.WriteString(`DELETE FROM zitadel.settings`)
+
+	writeCondition(&builder, condition)
+	return client.Exec(ctx, builder.String(), builder.Args()...)
+}
+
+func (s *settings) CheckMandatoryCondtions(condition database.Condition) error {
+	if !condition.IsRestrictingColumn(s.InstanceIDColumn()) {
+		return database.NewMissingConditionError(s.InstanceIDColumn())
+	}
+	if !condition.IsRestrictingColumn(s.OrganizationIDColumn()) {
```

Comment:
Is the need for an organizationID set outside of the repository? For example if you want to have a specific organization setting?